cli/command.go

```diff
@@ -274,19 +274,46 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, args ...any) (string, error) {
+	resultCh := make(chan string, 1)
+	errCh := make(chan error, 1)
 
-	if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
-		fmt.Fprint(c.Stdout(), msg)
-	}
+	// Prompts for user input and returns the value or an error if there is one.
+	go func() {
+		defer close(resultCh)
+		defer close(errCh)
+
+		scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+
+		if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
+			fmt.Fprint(c.Stdout(), msg, args)
+		}
 
-	scanner.Scan()
+		scanner.Scan()
 
-	if err := scanner.Err(); err != nil {
-		return "", fmt.Errorf("failed to read stdin: %w", err)
+		if err := scanner.Err(); err != nil {
+			select {
+			case errCh <- fmt.Errorf("failed to read stdin: %w", err):
+			default:
+			}
+		}
+
+		select {
```

Comment:
Can you help me understand why we need `select` here? Can it be just `resultCh <- scanner.Text()`?

Reply:
See comment from Seth: https://github.com/abcxyz/pkg/pull/143#discussion_r1242225484

Reply:
But the channel is buffered, so it can't block. The channel's capacity is 1, and there's only ever 1 send, so no blocking can happen. The select can be removed.

But if you take my suggestion above for a simpler way to write this function, this whole thing goes away.